### PR TITLE
Standalone and base EHR updates

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
In conjunction with build-outs for the two most recent EHR projects, we're creating a base EHR which can be used as a standalone EHR project or a base EHR module for a custom build-out but with more focused scope of functionality. This adds a base study and base set of reports.  EHR admin page update to allow more automatic functionality to avoid copying whole files of code for small changes. Start moving away from ExtJS and LDK UIs.

#### Related Pull Requests
* https://github.com/LabKey/nircEHRModules/pull/181
* https://github.com/LabKey/johnsHopkinsEHRModules/pull/1

#### Changes
* EHRCustomModule module property added
* Load Reports API - loads base reports from EHR first then additional reports from custom module
* Load Lookups API - will reference the new lookupsManifest file in either the custom module or EHR module for ehr lookups
* New UI for loading Reports and Lookups
* New UI for loading extensible column templates
* New EHR admin page UI
* Standalone EHR reference study
* Base set of reports

### CODE REVIEW

File: ehr/resources/views/ehrAdmin.html [hunk 1]
```diff
@@ -1,96 +1,44 @@
...
-Ext4.onReady(function (){
+</script>
 
-if (LABKEY.Security.currentUser.isAdmin){
-    var webpart = <%=webpartContext%>;
+<html>
+<div class="col-lg-3">
+<div class="lead">
+    <h4><b>Admin:</b></h4>
+    <div><a class="labkey-text-link" href='<%=contextPath%><%=containerPath%>/study-manageStudy.view?'>Manage EHR Study</a></div>
```
Comment (labkey-jeckels) [inline ehr/resources/views/ehrAdmin.html]: I know this syntax works, but I always prefer to use double quotes in HTML for attributes.
```suggestion
    <div><a class="labkey-text-link" href="<%=contextPath%><%=containerPath%>/study-manageStudy.view?">Manage EHR Study</a></div>
```

File: ehr/resources/views/ehrAdmin.html [hunk 2]
```diff
@@ -1,96 +1,44 @@
...
+    <div><a class="labkey-text-link" href='<%=contextPath%><%=containerPath%>/ehr-populateLookupData.view?'>Populate Reports and Lookups</a></div>
+    <div><a class="labkey-text-link" href='<%=contextPath%><%=containerPath%>/core-modulePropertyAdmin.view?moduleName=EHR'>Set EHR Properties</a></div>
+    <div><a class="labkey-text-link" href='<%=contextPath%><%=containerPath%>/ehr-ensureQcStates.view?'>Add QC States</a></div>
+    <div><a class="labkey-text-link" href='<%=contextPath%><%=containerPath%>/ehr-ehrTemplates.view?'>EHR Extensible Columns</a></div>
+    <div><a class="labkey-text-link" href='<%=contextPath%>/ehr<%=containerPath%>/primeDataEntryCache.view?'>Prime/Refresh EHR Data Cache</a></div>
```
Comment (labkey-jeckels) [inline ehr/resources/views/ehrAdmin.html]: Can we use the "new" URL syntax consistently here? For example:

```suggestion
    <div><a class="labkey-text-link" href='<%=contextPath%><%=containerPath%>/ehr-primeDataEntryCache.view?'>Prime/Refresh EHR Data Cache</a></div>
```

File: ehr/resources/views/ehrAdmin.html [hunk 3]
```diff
@@ -1,96 +1,44 @@
 <script type="text/javascript">
```
Comment (labkey-jeckels) [inline ehr/resources/views/ehrAdmin.html]: Do any current EHR modules use this implementation? If so, they probably won't want to start showing admin links to all users. But if not, then it's fine.

**Review by labkey-jeckels — APPROVED** [review]

File: ehr/resources/views/ehrAdmin.html (hunk 1 above)
Comment (labkey-martyp) [reply ehr/resources/views/ehrAdmin.html]: Updated

File: ehr/resources/views/ehrAdmin.html (hunk 2 above)
Comment (labkey-martyp) [reply ehr/resources/views/ehrAdmin.html]: Fixed all links using the old URL style

File: ehr/resources/views/ehrAdmin.html (hunk 3 above)
Comment (labkey-martyp) [reply ehr/resources/views/ehrAdmin.html]: All other EHR clients override this view.  The view.xml also sets permission to admin.

**Pull request merged**